### PR TITLE
Add example to `optuna.study.Study.stop` 

### PR DESCRIPTION
**Pull request by Ruketa**
## Motivation
This PR is related to #1334

## Description of the changes
Add the example for optuna.study.Study.stop

### CODE REVIEW

File: optuna/study.py [hunk 1]
```diff
@@ -481,6 +481,23 @@ def stop(self) -> None:
...
+
+                study = optuna.create_study()
+
+                def objective(trial):
+                    if trial.number == 4:
+                        study.stop()
+                    x = trial.suggest_uniform("x", 0, 10)
+                    return x ** 2
+
+                study.optimize(objective, n_trials=10)
```
Comment (hvy) [inline optuna/study.py:498]: A nit. Not really a convention but for consistency with other code examples.
```suggestion
                def objective(trial):
                    if trial.number == 4:
                        study.stop()
                    x = trial.suggest_uniform("x", 0, 10)
                    return x ** 2
                    
                study = optuna.create_study()
                study.optimize(objective, n_trials=10)
```

File: optuna/study.py [hunk 2]
```diff
@@ -481,6 +481,23 @@ def stop(self) -> None:
...
+                assert len(study.trials) == 5
```
Comment (hvy) [inline optuna/study.py:499]: Great that you added an assertion demonstrating the effects of `stop`. 👍

**Review by hvy — CHANGES_REQUESTED** [review]
Thanks for the PR. The usage of `stop` isn't obvious so this'll absolutely help users out. I left one minor comment.

This is by the way just a comment and not a change request but it might be more interesting to stop on the optimization based on the objective value rather than the number of trials (since this can be specified in `optimize`).

**Review by HideakiImamura — APPROVED** [review]
Thanks for the PR! LGTM except for @hvy's comment.

File: optuna/study.py (hunk 1 above)
Comment (Ruketa) [reply optuna/study.py]: Thank you for reviewing PR. I fixed it.

**Review by hvy — APPROVED** [review]
LGTM, thank you!

**Pull request merged**